server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/cassandra/FirstEnqueuedMailDao.java

```diff
@@ -0,0 +1,74 @@
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.projection.cassandra;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static org.apache.james.queue.rabbitmq.projection.cassandra.FirstEnqueuedMailTable.QUEUE_NAME;
+import static org.apache.james.queue.rabbitmq.projection.cassandra.FirstEnqueuedMailTable.TABLE_NAME;
+import static org.apache.james.queue.rabbitmq.projection.cassandra.FirstEnqueuedMailTable.TIME_RANGE_START;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.queue.rabbitmq.projection.cassandra.model.FirstEnqueuedItem;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+
+public class FirstEnqueuedMailDao {
+
+    private final CassandraAsyncExecutor executor;
+    private final PreparedStatement select;
+    private final CassandraUtils cassandraUtils;
+
+    @Inject
+    public FirstEnqueuedMailDao(Session session, CassandraUtils cassandraUtils) {
+        this.executor = new CassandraAsyncExecutor(session);
+        this.cassandraUtils = cassandraUtils;
+
+        this.select = prepareSelect(session);
+    }
+
+    private PreparedStatement prepareSelect(Session session) {
+        return session.prepare(select()
+                .from(TABLE_NAME)
+                .where(eq(QUEUE_NAME, bindMarker(QUEUE_NAME))));
+    }
+
+    public CompletableFuture<Optional<FirstEnqueuedItem>> findFirstEnqueuedItem(String queueName) {
+        return executor.executeSingleRow(
+            select.bind()
+                .setString(QUEUE_NAME, queueName))
+            .thenApply(optional -> optional.map(this::toPojo));
+    }
+
+    private FirstEnqueuedItem toPojo(Row row) {
```

Comment:
No you don't return the first enqueued item, but it's timestamp. It is thus not `FirstEnqueuedItem`...